idaes/apps/caprese/estimator.py

```diff
@@ -0,0 +1,597 @@
+                                               solver):
+        '''
+        Before running MHE, we need to set up past information (measurements, states, etc.). 
+        Here we solve for a steady state and set it as past information.
+
+        Parameters
+        ----------
+        desired_ss : List of vardata, value tuples describing the
+                     desired steady stsates of these specified variables.
+        ss_weights : List of vardata, value tuples describing the
+                     weightss of these specified variables.
+
+        '''
+        self.add_steady_state_objective(desired_ss, ss_weights)
+        self.solve_steady_state(solver)
+
+        self.initialize_actualmeasurements_at_t0()
+        self.initialize_to_initial_conditions(ctype=(DiffVar, AlgVar, DerivVar, InputVar,))
+
+    def add_noise_minimize_objective(self,
+                                     model_disturbance_weights,
+                                     measurement_noise_weights,
+                                     givenform = "weight"):
+        '''
+        Set up the objective function for MHE. The given form can be either 'weight'
+        or 'variance'.
+
+        Parameters
+        ----------        
+        model_disturbance_weights: A list of vardata-value tuples describing 
+                                    the weight for model disturbance to be 
+                                    given to the objective term containing each variable.
+        measurement noise_weights: A list of vardata-value tuples describing 
+                                    the weight for measurement error to be 
+                                    given to the objective term containing each variable.
+        givenform: The form of given lists. It should be either "weight" (default)
+                    or "variance".
+        '''
+
+        if givenform not in ["weight", "variance"]:
+            raise RuntimeError("Wrong argument 'givenform' is given. "
+                   "Please assign either 'weight' or 'variance'.")
+
+        diff_id_list = [id(var[0]) for var in self.differential_vars]
+        for var, val in model_disturbance_weights:  
+            #Check whether the given variable is classified under diffvar
+            if id(var) not in diff_id_list:
+                raise RuntimeError(var.name, 
+                                   " is not a differential variable.")
+
+            if givenform == "weight":
+                self.diffvar_map_moddis[var].weight = val
+            elif givenform == "variance":
+                self.diffvar_map_moddis[var].weight = 1./val
+
+        mea_id_list = [id(var[0]) for var in self.measurement_vars]
+        for var, val in measurement_noise_weights:
+            #Check whether the given variable is declared as a measurement before
+            if id(var) not in mea_id_list:
+                raise RuntimeError(var.name, 
+                                   " is not declared as a measurement.")
```

Comment:
Same comment as above regarding `0` vs `t0` and checking inclusion in a list vs. set.

Reply:
thanks!